src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -522,50 +524,69 @@ public SCM build(SCMHead head, SCMRevision revision) {
                 }
             }
         }
+        String browserUrl = buildBitbucketClient().getRepositoryUri(repositoryType, BitbucketRepositoryProtocol.HTTP, null, repoOwner, repository);
         if (head instanceof PullRequestSCMHead) {
             PullRequestSCMHead h = (PullRequestSCMHead) head;
             if (repositoryType == BitbucketRepositoryType.MERCURIAL) {
                 MercurialSCM scm = new MercurialSCM(getRemote(h.getRepoOwner(), h.getRepository(),
                         BitbucketRepositoryType.MERCURIAL));
                 // If no revision specified the branch name will be used as revision
-                scm.setRevision(revision instanceof MercurialRevision
-                        ? ((MercurialRevision) revision).getHash()
-                        : h.getBranchName()
-                );
-                scm.setRevisionType(RevisionType.BRANCH);
+                if (revision instanceof MercurialRevision) {
```

Comment:
It does seem like there are two distinct bugs being fixed here, though there may be a dependency between them.

Is there no test coverage in this plugin?